src/test/ensemble_tests.cpp

```diff
@@ -2195,3 +2195,146 @@ TEST_F(EnsembleFlowTest, WaitForLoadingPipelineDefinitionFromBeginStatus) {
+            }
+        }
+    ]
+})";
+
+TEST_F(EnsembleFlowTest, RetirePipelineAfterLoading) {
+    std::string fileToReload = "/tmp/ovms_config_file1.json";
+    createConfigFileWithContent(pipelineOneDummyConfig, fileToReload);
+    ConstructorEnabledModelManager manager;
+    auto status = manager.startFromFile(fileToReload);
+    manager.startWatcher();
+    ASSERT_TRUE(status.ok()) << status.string();
+    ASSERT_EQ(manager.getPipelineFactory().findDefinitionByName(PIPELINE_1_DUMMY_NAME)->getStatusCode(),
+        PipelineDefinitionStateCode::AVAILABLE);
+    createConfigFileWithContent(configJsonWithNoPipeline, fileToReload);
+    std::this_thread::sleep_for(std::chrono::milliseconds(1500));
+}
+const char* pipelineOneDummyConfigWithChangedInputName = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"all": {}},
+                "nireq": 1
+            }
+        }
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "pipeline1Dummy",
+            "inputs": ["NEW_INPUT_NAME"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                               "data_item": "NEW_INPUT_NAME"}}
+                    ],
+                    "outputs": [
+                        {"data_item": "a",
+                         "alias": "new_dummy_output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                         "data_item": "new_dummy_output"}
+                }
+            ]
+        }
+    ]
+})";
+const std::string NEW_INPUT_NAME = "NEW_INPUT_NAME";
+
+TEST_F(EnsembleFlowTest, ReloadPipelineAfterLoadingSuccesfullyChangedInputName) {
+    std::string fileToReload = "/tmp/ovms_config_file1.json";
+    createConfigFileWithContent(pipelineOneDummyConfig, fileToReload);
+    ConstructorEnabledModelManager manager;
+    auto status = manager.startFromFile(fileToReload);
+    manager.startWatcher();
+    ASSERT_TRUE(status.ok()) << status.string();
+    ASSERT_EQ(manager.getPipelineFactory().findDefinitionByName(PIPELINE_1_DUMMY_NAME)->getStatusCode(),
+        PipelineDefinitionStateCode::AVAILABLE);
+
+    tensor_map_t inputsInfoBefore;
+    auto pdPtr = manager.getPipelineFactory().findDefinitionByName(PIPELINE_1_DUMMY_NAME);
+    status = pdPtr->getInputsInfo(inputsInfoBefore, manager);
+    ASSERT_TRUE(status.ok()) << status.string();
+    ASSERT_EQ(inputsInfoBefore.count(NEW_INPUT_NAME), 0);
+
+    // now reload
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
These constant sleep intervals depend on watcher interval time. In the future, when somebody changes default watcher sleep from 1s to 10s all these tests that rely on sleeping will fail.